scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerBackupRunner.java

```diff
@@ -108,16 +111,18 @@ private void performBackup() throws IOException {
 
     private void removeOldBackups() {
         File fodlerWhereBackups = new File(destination);
-        Stream.of(Objects.requireNonNull(fodlerWhereBackups.listFiles()))
-              .filter(File::isFile)
-              .filter(file -> file.getName().startsWith(PREFIX))
-              .sorted(Comparator.comparingLong(file -> getCreationTime((File) file)).reversed())
-              .skip(windowSize - 1)
-              .forEach(file -> {
-                  LOGGER.info("Removing old backup: " + file.getName());
-                  file.delete();
-              });
-
+        File[] files = fodlerWhereBackups.listFiles();
+        if (files != null) {
+            Stream.of(files)
+                  .filter(File::isFile)
+                  .filter(file -> file.getName().startsWith(PREFIX))
```

Comment:
As explained here https://dzone.com/articles/single-filter-perform-better-than-multiple-one-in, maybe it is better to use a single filter instead of many.

Reply:
Thanks Amine, I did not know it. However in this case I think performance improvement is really neglactable compairng to code clarity. I will leave as it is. Sorry